test/decoders/test_ops.py

```diff
@@ -628,22 +626,179 @@ def test_cuda_decoder(self):
+        decoder = create_from_file(str(NASA_AUDIO.path), seek_mode="approximate")
         add_audio_stream(decoder)
         with pytest.raises(RuntimeError, match="The method you called isn't supported"):
             method(decoder)
 
     def test_audio_bad_seek_mode(self):
-        decoder = create_from_file(str(NASA_AUDIO_MP3.path), seek_mode="exact")
+        decoder = create_from_file(str(NASA_AUDIO.path), seek_mode="exact")
         with pytest.raises(
             RuntimeError, match="seek_mode must be 'approximate' for audio"
         ):
             add_audio_stream(decoder)
 
+    @pytest.mark.parametrize(
+        "range",
+        (
+            "begin_to_end",
+            "begin_to_None",
+            "begin_to_beyond_end",
+            "at_frame_boundaries",
+            "not_at_frame_boundaries",
+        ),
+    )
+    @pytest.mark.parametrize("asset", (NASA_AUDIO, NASA_AUDIO_MP3))
+    def test_get_frames_by_pts_in_range_audio(self, range, asset):
+        if range == "begin_to_end":
+            start_seconds, stop_seconds = 0, asset.duration_seconds
+        elif range == "begin_to_None":
+            start_seconds, stop_seconds = 0, None
+        elif range == "begin_to_beyond_end":
+            start_seconds, stop_seconds = 0, asset.duration_seconds + 10
+        elif range == "at_frame_boundaries":
+            start_seconds = asset.get_frame_info(idx=10).pts_seconds
+            stop_seconds = asset.get_frame_info(idx=40).pts_seconds
+        else:
+            assert range == "not_at_frame_boundaries"
+            start_frame_info = asset.get_frame_info(idx=10)
+            stop_frame_info = asset.get_frame_info(idx=40)
+            start_seconds = start_frame_info.pts_seconds + (
+                start_frame_info.duration_seconds / 2
+            )
+            stop_seconds = stop_frame_info.pts_seconds + (
+                stop_frame_info.duration_seconds / 2
+            )
+
+        ref_start_index = asset.get_frame_index(pts_seconds=start_seconds)
+        if range == "begin_to_None":
+            ref_stop_index = (
+                asset.get_frame_index(pts_seconds=asset.duration_seconds) + 1
+            )
+        elif range == "at_frame_boundaries":
+            ref_stop_index = asset.get_frame_index(pts_seconds=stop_seconds)
+        else:
+            ref_stop_index = asset.get_frame_index(pts_seconds=stop_seconds) + 1
+        reference_frames = asset.get_frame_data_by_range(
+            start=ref_start_index,
+            stop=ref_stop_index,
+        )
+
+        decoder = create_from_file(str(asset.path), seek_mode="approximate")
+        add_audio_stream(decoder)
+
+        frames = get_frames_by_pts_in_range_audio(
+            decoder, start_seconds=start_seconds, stop_seconds=stop_seconds
+        )
+
+        torch.testing.assert_close(frames, reference_frames)
+
+    @pytest.mark.parametrize(
+        "asset, expected_shape", ((NASA_AUDIO, (2, 1024)), (NASA_AUDIO_MP3, (2, 576)))
```

Comment:
Can we use the number of channels and sample rate from the `TestContainer`? I know that for MP3, this is potentially a problem as the sample rate tends to vary, but noting that in a comment would help as well.